mentat/diff_context.py

```diff
@@ -133,6 +133,8 @@ def __init__(
 
         meta = get_treeish_metadata(git_root, target)
         name += f'{meta["hexsha"][:8]}: {meta["summary"]}'
+        if target == "HEAD":
+            name = "HEAD (last commit)"
```

Comment:
If you take this line out, the name would be like `HEAD: 2ab23bc8: add default args for diff`. To me that's preferable to just `HEAD (last commit)` if I've added the `--diff` flag.

Reply:
Hmm I see what you're saying, but the name doesn't have HEAD at the start; it's just ```c7b6123b: add conversation tokens and use real token counts in cost tracker (#283) | 1 files | 2 lines``` which I find a lot worse than ```HEAD (last commit)``` (since it doesn't specify that it's HEAD)

Reply:
I vote with Grant. They should know it's diffed with head when they ran `--diff`. But they may not know the file/line counts. Maybe we could split the difference with `name += " (HEAD)"` like it appears in `git log`?

Reply:
Oh the file and line counts are always there for both; and they might not know it's diffed with HEAD when they run just --diff (because HEAD is the default). I could be ok with adding (HEAD) to the end, but I still think it makes more sense to not have the commit hash or description there; after all, this is a diff with HEAD, so it's just the current working changes.

Reply:
You've convinced me. I guess I don't feel that strongly one way or the other anyway. One more open would be to write just HEAD since they should no that means it's diffed with the last commit. But all the options seem fine.

Reply:
First off, I wasn't happy with how get_code_message was handling this, so in #291 I have this update in CodeContext:
```
        diff = self.diff_context.get_display_context()
        if diff:
            stream.send(f"{prefix}Diff: {diff}", color="green")
```
So if there are no files, the Diff display is omitted altogether.

After reading you guys' convo, I think we should omit the hexsha/description altogether, and just display `<target> | <i> files | <i> lines`. This is concise, and if they want to see details, they can copy and run a git diff in the terminal.